modules/lib/src/main/resources/assets/js/app/wizard/WorkflowStateManager.ts

```diff
@@ -46,62 +40,54 @@ export class WorkflowStateManager {
     private updateIcons(): void {
         const thumbnailUploader: ThumbnailUploaderEl = this.wizard.getFormIcon();
         const toolbarIcon: DivEl = this.wizard.getMainToolbar()?.getStateIcon();
+        const isStatusHidden: boolean = this.isStatusToBeHidden();
 
-        if (thumbnailUploader) {
-            this.toggleWorkflowStateClasses(thumbnailUploader);
-        }
+        thumbnailUploader.setStatus(this.status);
+        thumbnailUploader.toggleClass('status-hidden', isStatusHidden);
 
-        if (toolbarIcon) {
-            this.toggleWorkflowStateClasses(toolbarIcon);
-        }
+        toolbarIcon?.toggleClass(WorkflowStateStatus.INVALID, this.status === WorkflowStateStatus.INVALID);
+        toolbarIcon?.toggleClass(WorkflowStateStatus.READY, this.status === WorkflowStateStatus.READY);
+        toolbarIcon?.toggleClass(WorkflowStateStatus.IN_PROGRESS, this.status === WorkflowStateStatus.IN_PROGRESS);
```

Comment:
`toolbarIcon?.toggleClass(WorkflowStateStatus.INVALID, WorkflowStateStatus.isInvalid(this.status));`

etc.